rules_128tech/pylint_plugins/mock_patch_autospec.py

```diff
@@ -0,0 +1,69 @@
+"""
+A plugin to detect and disallow the use of mock.patch or mocker.patch without
+setting autospec=True.
+
+Based on http://pylint.pycqa.org/en/latest/how_tos/custom_checkers.html#write-a-checker
+"""
+
+import astroid
+import astroid.node_classes
+from pylint.checkers import BaseChecker
+from pylint.interfaces import IAstroidChecker
+
+
+def register(linter):
+    linter.register_checker(MockPatchAutospecChecker(linter))
+
+
+class MockPatchAutospecChecker(BaseChecker):
+    """A pylint checker to detect use of mock.patch without autospec=True"""
+
+    __implements__ = IAstroidChecker
+
+    name = "mock-patch-autospec"
+    priority = -1
+    msgs = {
+        "W1280": (
+            "Uses mock.patch without autospec=True.",
+            name,
+            (
+                "All mock patches should use autospec=True to avoid allowing calls "
```

Comment:
google names their pylint checks starting with `g-`... I wonder if we should name ours starting with `128t-`, or at least have some hint in the message that this is a custom checker. Otherwise if someone saw this error and tried to search the official docs they'd have a bad time